### PR TITLE
Changes to enable HPA for ama-metrics deployment

### PR DESCRIPTION
**Pull request by rashmichandrashekar**
[comment]: # (Note that your PR title should follow the conventional commit format: https://conventionalcommits.org/en/v1.0.0/#summary)
# PR Description
Changes to enable HPA for ama-metrics deployment.
Update ME version to include changes for memory deallocation
Updating ME config to tune memory deallocation
Telemetry to track HPA enablement

[comment]: # (The below checklist is for PRs adding new features. If a box is not checked, add a reason why it's not needed.)
# New Feature Checklist

- [X] List telemetry added about the feature - Added CollectorHpaEnabled in the custom dimensions
- [X] Link to the one-pager about the feature - https://msazure.visualstudio.com/InfrastructureInsights/_wiki/wikis/InfrastructureInsights.wiki/686240/HPA-PRD

### CODE REVIEW

**Review by gracewehner — APPROVED** [review]

File: .pipelines/azure-pipeline-build.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ trigger:
   branches:
     include:
     - main
+    - rashmi/hpa
```
Comment (vishiy) [inline .pipelines/azure-pipeline-build.yml:5]: pls remove branch...

File: otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/_ama-metrics-helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+{{/* If the current HPA already exists, set the HPA values to the current 
+     HPA spec to preserve those values. */}}
+
+{{- $amaMetricsCurrentHPA := lookup "autoscaling/v2" "HorizontalPodAutoscaler" "kube-system" $amaMetricsHpaName }}
+{{- if and $amaMetricsCurrentHPA $amaMetricsCurrentHPA.spec }}
+{{- $amaMetricsMinReplicasFromCurrentSpec := $amaMetricsCurrentHPA.spec.minReplicas -}}
+{{- $amaMetricsMaxReplicasFromCurrentSpec := $amaMetricsCurrentHPA.spec.maxReplicas -}}
+
+  {{- if and ($amaMetricsMinReplicasFromCurrentSpec) (gt (int $amaMetricsMinReplicasFromCurrentSpec) 0) }}
```
Comment (vishiy) [inline otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/_ama-metrics-helpers.tpl:24]: Not sure why we are comparing min from current spec , again with min from current spec ?

File: otelcollector/metricextension/me_ds_internal_win.config [hunk 3]
```diff
@@ -23,5 +23,7 @@
    "maxStringInternCacheSizeMb":5000,
    "interningSwapPeriodInMin":10000,
    "internalQueueSizeManagementPeriodInSec":10000,
-   "proxyDefinitionMode":1
+   "proxyDefinitionMode":1,
+   "memoryPoolReleasePeriodInMin":10,
+   "memoryPoolReleaseTriggerPercentage": 70
```
Comment (vishiy) [inline otelcollector/metricextension/me_ds_internal_win.config:28]: i see both time & percentage. so whatever comes first it triggers deallocation ? Also 70% is when the mem usage becomes 70% from the highest usage ?

Comment (rashmichandrashekar) [reply otelcollector/metricextension/me_ds_internal_win.config:28]: Basically, this means that the memory release evaluation is performed every 10 minutes. 
Yes your explanation for percentage is correct.

File: otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/_ama-metrics-helpers.tpl (hunk 2 above)
Comment (rashmichandrashekar) [reply otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/_ama-metrics-helpers.tpl:24]: The first check(line 24) is for checking if it is greater than 0 and the second(line 25) is to make sure it is greater than the minimum we want to support which is 2. 
Pls lmk if I got the lines wrong and explaining the wrong checks :)

**Review by vishiy — APPROVED** [review]

**Review by vishiy — APPROVED** [review]

**Pull request merged**